### PR TITLE
fix: timezone issues on notes

### PR DESCRIPTION
**Pull request by Rigerwal**
Use toDateString() method on dates in order to remove timezone to handle notes properly.
Create a list of all the dates on which we will loop instead of just looping on event days because there might be a difference between days present in note list and in post list because of this timezone difference

### CODE REVIEW

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 1]
```diff
@@ -149,12 +153,17 @@ export class CalendarMonthRowComponent implements OnChanges {
       .map((daySliced) =>
         this.timezoneManager.reverseTz(daySliced.date, this.timezone)
       )
-      .findIndex((date) => date.valueOf() === note.start.valueOf());
+      .findIndex(
+        (date) =>
+          new Date(date.toDateString()).valueOf() === note.start.valueOf()
```
Comment (HbDevelop) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:158]: date.toDateString() === note.start.toDateString() is better no ?

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 2]
```diff
@@ -149,12 +153,17 @@ export class CalendarMonthRowComponent implements OnChanges {
...
+      );
     const indexEndDate = this.daysSliced
       .map((daySliced) =>
         this.timezoneManager.reverseTz(daySliced.date, this.timezone)
       )
-      .findIndex((date) => date.valueOf() === note.end.valueOf());
+      .findIndex(
+        (date) => new Date(date.toDateString()).valueOf() === note.end.valueOf()
```
Comment (HbDevelop) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:165]: same here

**Review by Neewd — APPROVED** [review]

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts (hunk 1 above)
Comment (Rigerwal) [reply projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:158]: No need as note.start is already timezone free so note.start is equal to new Date(note.start.toDateString())

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts (hunk 2 above)
Comment (Rigerwal) [reply projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:165]: Same :)

**Review by HbDevelop — APPROVED** [review]

**Review by Njol75 — APPROVED** [review]

**Pull request merged**